subsys/bluetooth/controller/ll_sw/ull_adv_internal.h

```diff
@@ -129,5 +128,4 @@ uint32_t ull_adv_sync_start(struct ll_adv_sync_set *sync,
 
```

Comment:
The definition of `struct ll_adv_sync_set` in ull_adv_types.h is guarded with `CONFIG_BT_CTLR_ADV_PERIODIC`. Don't we need a forward declaration for a compilation without warnings?

Reply:
Yup, you are right. 

I guess this could be handled in different ways:
1) Guard the `ull_adv_sync_start` with the `CONFIG_BT_CTLR_ADV_PERIODIC` again
2) Remove the `CONFIG_BT_CTLR_ADV_PERIODIC` guard from `ll_adv_sync_set`
3) Forward declaration

I opted for 2) (removing a guard from a struct definition shouldn't have any side effects). 

@cvinayak Please let me know if you disagree with this approach